lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -115,91 +149,102 @@ sub get_data_from_parent_zone {
     my ( $self, $params ) = @_;
     my $domain = "";
 
-    if (ref \$params eq "SCALAR") {
-        $domain = $params;
-    } else {
-        $domain = $params->{"domain"};
-    }
-
     my %result;
+    eval {
+        if (ref \$params eq "SCALAR") {
+            $domain = $params;
+        } else {
+            $domain = $params->{"domain"};
+        }
 
-    my ( $dn, $dn_syntax ) = $self->_check_domain( $domain, 'Domain name' );
-    return $dn_syntax if ( $dn_syntax->{status} eq 'nok' );
+        my ( $dn, $dn_syntax ) = $self->_check_domain( $domain, 'Domain name' );
+        return $dn_syntax if ( $dn_syntax->{status} eq 'nok' );
```

Comment:
This won't make `get_data_from_parent_zone` return. It only terminates the eval block with $dn_syntax as return value.

Reply:
Fixed in dcf35a6